pebble/src/windows/main_window.c

```diff
@@ -0,0 +1,106 @@
+#include "main_window.h"
+
+#define SAMPLES_PER_UPDATE 5
+
+static Window *s_window;
+static TextLayer *s_text_layer;
+
+static AppTimer *s_packets_per_second_timer;
+
+static bool s_sending;
+static int s_send_counter, s_packets_per_second;
+
+static void packets_per_second_handler(void *context) {
+  s_packets_per_second = s_send_counter;
+  const int samples_per_second = s_packets_per_second * SAMPLES_PER_UPDATE;
+
+  if(s_sending) {
+    static char s_buff[32];
+    snprintf(s_buff, sizeof(s_buff), "%d packets/s\n(%d samples/s)", s_packets_per_second, samples_per_second);
+    text_layer_set_text(s_text_layer, s_buff);
+  }
+
+  s_send_counter = 0;
+  app_timer_register(1000, packets_per_second_handler, NULL);
+}
+
+static void send(AccelData *data, uint32_t num_samples) {
+  if(comm_send_data(data, num_samples)) {
+    window_set_background_color(s_window, GColorGreen);
+    s_send_counter++;
+  }
+}
+
+static void accel_data_handler(AccelData *data, uint32_t num_samples) {
+  // If ready to send
+  if(comm_is_busy()) {
+    APP_LOG(APP_LOG_LEVEL_WARNING, "Accel sample arrived early");
+    window_set_background_color(s_window, GColorRed);
+  } else {
+    send(data, num_samples);
+  }
+}
+
+static void select_click_handler(ClickRecognizerRef recognizer, void *context) {
+  if(!s_sending) {
+    // Begin sending data
+    s_sending = true;
+
+    accel_service_set_sampling_rate(ACCEL_SAMPLING_25HZ);
+    accel_data_service_subscribe(SAMPLES_PER_UPDATE, accel_data_handler);
+
+    text_layer_set_text(s_text_layer, "Started");
+    if(s_packets_per_second_timer) {
+      app_timer_cancel(s_packets_per_second_timer);
+    }
+    s_packets_per_second_timer = app_timer_register(1000, packets_per_second_handler, NULL);
+    window_set_background_color(s_window, GColorOrange);
+  } else {
+    // Stop sending data
+    s_sending = false;
+
+    accel_data_service_unsubscribe();
+
+    text_layer_set_text(s_text_layer, "Stopped");
+    if(s_packets_per_second_timer) {
+      app_timer_cancel(s_packets_per_second_timer);
+      s_packets_per_second_timer = NULL;
+    }
+    window_set_background_color(s_window, GColorDarkGray);
+  }
+}
+
+static void click_config_provider(void *context) {
+  window_single_click_subscribe(BUTTON_ID_SELECT, select_click_handler);
+}
+
+static void window_load(Window *window) {
+  Layer *window_layer = window_get_root_layer(window);
+  GRect bounds = layer_get_bounds(window_layer);
+
+  s_text_layer = text_layer_create(GRect(bounds.origin.x, bounds.origin.y + 50, bounds.size.w, bounds.size.h));
```

Comment:
I'm not super concerned about this, since nothing is beneath it. Let me know if you feel strongly.